lms/djangoapps/courseware/views/views.py

```diff
@@ -707,6 +708,63 @@ def course_about(request, course_id):
         return render_to_response('courseware/course_about.html', context)
 
 
+# TODO: make this efficient
+def _add_course_info(program, user):
+    for program_course in program['courses']:
+        for course_run in program_course['course_runs']:
+            course = modulestore().get_course(CourseKey.from_string(course_run['key']))
+            course_run.update({
+                'registered': registered_for_course(course, user),
+                'can_enroll': bool(has_access(user, 'enroll', course)),
+                'is_shib_course': uses_shib(course)
+            })
+
+
+def _get_youtube_url_identifier(url):
+    pattern = r'(?:youtube\.com\/(?:[^\/]+\/.+\/|(?:v|e(?:mbed)?)\/|.*[?&]v=)|youtu\.be\/)([^"&?\/ ]{11})'
+    patterns = re.search(pattern, url)
+    if patterns:
+        return patterns.groups()[0]
+    return url
+
+
+@ensure_csrf_cookie
+@cache_if_anonymous()
+def program_marketing(request, program_uuid):
+    """
+    Display the program marketing page.
+    """
+    program_data = get_programs(uuid=program_uuid)
+
+    if not program_data:
+        raise Http404
+
+    program_data = ProgramDataExtender(program_data, request.user).extend(include_instructors=True)
```

Comment:
FYI: Renzo made me aware of this `ProgramDataExtender` class. We should use it to add additional data to the program dict that we may need for rendering the program details page instead of the `_add_course_info` function we have created above.